src/purple_vm/src/primitives/value.rs

```diff
@@ -149,6 +151,22 @@ impl VmValue {
         }
     }
 
+    pub fn i32trunc_signedf32(&self) -> Result<VmValue, VmError> {
+        match *self {
+            VmValue::F32(val) => {
+                println!("before cast {:?} -> {:?}", val, val.to_be_bytes());
```

Comment:
Please avoid using `println!()` statements as much as you can. While they are useful for debugging one should be very careful of leaking them into the `master` branch as they bypass any logging front-end and can result in performance bottlenecks. I suggest using one either the `trace!()` or `debug!()` macros.